frontend/src/api/ApiProvider.ts

```diff
@@ -0,0 +1,41 @@
+import { 
+  BASE_API_URL
+} from '@/helpers/constants';
+
+interface IHeaders {
+  [n: string]: string
+};
+
+export const createRequest = async (
+    endpoint: string, 
+    method: string, 
+    data: object | null = null
+  ) => {
+  const url = `${BASE_API_URL}/${endpoint}`;
+  const headers: IHeaders = {
+    'Content-Type': 'application/json'
+  };
+  
+  const token = document.cookie;
+
+  if (token) {
+    headers['Authorization'] = `Bearer ${token.substring(6)}`;

```

Comment:
substring(6) is a very unreliable way to get a substring

Reply:
Done